builtin/providers/cidr/datasource_cidr_network_test.go

```diff
@@ -0,0 +1,341 @@
+				&subnetMask{Name: "nat", Mask: 29},
+				&subnetMask{Name: "4_special", Mask: 22},
+			},
+		},
+		&testCase{
+			CurrentSubnet: "255.255.252.0/24",
+			Zones:         1,
+			SubnetMasks: []*subnetMask{
+				&subnetMask{Name: "private", Mask: 27},
+				&subnetMask{Name: "public", Mask: 24},
+				&subnetMask{Name: "nat", Mask: 29},
+				&subnetMask{Name: "4_special", Mask: 22},
+			},
+		},
+	}
+	testCases := []*testCase{
+		&testCase{
+			CurrentSubnet: "9.255.255.0/24",
+			Zones:         1,
+			SubnetMasks: []*subnetMask{
+				&subnetMask{Name: "private", Mask: 24},
+				&subnetMask{Name: "public", Mask: 25},
+				&subnetMask{Name: "nat", Mask: 26},
+				&subnetMask{Name: "4_special", Mask: 27},
+			},
+			CIDRList: []string{
+				"10.0.0.0/24",
+				"10.0.1.0/25",
+				"10.0.1.128/26",
+				"10.0.1.192/27",
+			},
+			CIDRByName: map[string]string{
+				"private":   "10.0.0.0/24",
+				"public":    "10.0.1.0/25",
+				"nat":       "10.0.1.128/26",
+				"4_special": "10.0.1.192/27",
+			},
+		},
+		&testCase{
+			CurrentSubnet: "192.168.7.0/24",
+			Zones:         1,
+			SubnetMasks: []*subnetMask{
+				&subnetMask{Name: "private", Mask: 24},
+				&subnetMask{Name: "public", Mask: 25},
+				&subnetMask{Name: "nat", Mask: 26},
+				&subnetMask{Name: "4_special", Mask: 27},
+			},
+			CIDRList: []string{
+				"192.168.8.0/24",
+				"192.168.9.0/25",
+				"192.168.9.128/26",
+				"192.168.9.192/27",
+			},
+			CIDRByName: map[string]string{
+				"private":   "192.168.8.0/24",
+				"public":    "192.168.9.0/25",
+				"nat":       "192.168.9.128/26",
+				"4_special": "192.168.9.192/27",
+			},
+		},
+		&testCase{
+			CurrentSubnet: "2001:db8:c001:b9c0::/58",
+			Zones:         1,
+			SubnetMasks: []*subnetMask{
+				&subnetMask{Name: "private", Mask: 58},
+				&subnetMask{Name: "public", Mask: 58},
+				&subnetMask{Name: "nat", Mask: 58},
+				&subnetMask{Name: "4_special", Mask: 58},
+			},
+			CIDRList: []string{
+				"2001:db8:c001:ba00::/58",
+				"2001:db8:c001:ba40::/58",
+				"2001:db8:c001:ba80::/58",
+				"2001:db8:c001:bac0::/58",
+			},
+			CIDRByName: map[string]string{
+				"private":   "2001:db8:c001:ba00::/58",
+				"public":    "2001:db8:c001:ba40::/58",
+				"nat":       "2001:db8:c001:ba80::/58",
+				"4_special": "2001:db8:c001:bac0::/58",
+			},
+		},
+	}
+
+	for _, tc := range failCases {
+		_, IPNet, err := net.ParseCIDR(tc.CurrentSubnet)
+		if err != nil {
+			t.Errorf("Failed to parse %s\n", tc.CurrentSubnet)
+		}
+		_, _, fail := calculateSubnets(IPNet, tc.SubnetMasks)
+		if fail == nil {
+			t.Errorf("calculateSubnets shoudl have failed starting at %s\n", tc.CurrentSubnet)
+		}
+	}
+	for _, tc := range testCases {
+		_, IPNet, err := net.ParseCIDR(tc.CurrentSubnet)
+		if err != nil {
+			t.Errorf("Failed to parse %s\n", tc.CurrentSubnet)
+		}
+		cByName, _, err := calculateSubnets(IPNet, tc.SubnetMasks)
+		if err != nil {
+			t.Errorf("failed calculateSubnets with %v\n", err)
```

Comment:
I'd rewrite this as:

calculateSubnets failed with...

Reply:
fixed in https://github.com/moofish32/terraform-provider-cidr/pull/2